files/initramfs-tools/union-mount.j2

```diff
@@ -39,10 +39,49 @@ set_tmpfs_log_partition_size()
   [ $maxsize -le $varlogsize ] && varlogsize=$maxsize
 }
 
+whitelist_rw_folder()
+{
+  image_dir=$1
+  whitelist_file=${rootmnt}/host/$image_dir/whitelist
+ 
+  # Return if the whitelist file does not exist
+  if ! test -f "${whitelist_file}"; then
+    return
+  fi
+ 
+  # Return if the secure_boot_enable option is not set
+  if cat /proc/cmdline | grep -v -q "secure_boot_enable=true"; then
+    return
+  fi
+ 
+  whitelist_log=${rootmnt}/host/$image_dir/whitelist.log
+  rw_dir=${rootmnt}/host/$image_dir/rw
+  whitelist=$(cat ${rootmnt}/host/$image_dir/whitelist | grep -v "^\s*#" | awk '{$1=$1};1')
```

Comment:
> | grep -v "^\s*#" | awk '{$1=$1};1' [](start = 54, length = 36)

Is it only for parsing comments? If yes, we may remove the complexity. And add a `whiltelist_README.md` side-by-side to explain usage.

Reply:
Offline discussed, move comment out into separate file. It is not a runtime conf file.

---
In reply to: [424122684](https://github.com/Azure/sonic-buildimage/pull/4585#discussion_r424122684) [](ancestors = 424122684)